play-services-location/core/src/main/kotlin/org/microg/gms/location/settings/LocationSettingsCheckerActivity.kt

```diff
@@ -141,7 +140,22 @@ class LocationSettingsCheckerActivity : Activity(), DialogInterface.OnCancelList
 
     private fun handleContinue() {
         val improvement = improvements.firstOrNull() ?: return finishResult(RESULT_OK)
+        displayList.add(improvement)
         when (improvement) {
+            Improvement.PERMISSIONS -> {
+                val targetIntent = if (intent.hasExtra(EXTRA_SHOW_MG_SETTINGS) && intent.getBooleanExtra(EXTRA_SHOW_MG_SETTINGS, false)) {
+                    Intent().apply {
+                        action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+                        data = Uri.fromParts("package", Constants.GMS_PACKAGE_NAME, null)
+                    }
+                }else{
```

Comment:
Decision to show application details settings should happen in `AskPermissionActivity`, it's not specific to the `LocationSettingsCheckerActivity`.